File: #2348.number-of-zero-filled-subarrays.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        
        int size = nums.size();
        long long count = 0;
        long long result = 0;
        for (int i = 0; i < size; i++) {
            count = nums[i] == 0 ? count + 1 : 0;
            result += count;
        }
        return result;
    }
};

File: #2348.number-of-zero-filled-subarrays.ts
function zeroFilledSubarray(nums: number[]): number {
    let j = -1;
    let zeroes = 0;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i]) {
            j = i;
        } else {
            zeroes += i - j;
        }
    }
    return zeroes;
};
